R/dataset.R

```diff
@@ -288,7 +288,7 @@ Dataset.get_by_full_path <- function(full_path) {
     object = Object.get_by_full_path(full_path)
 
     if (is.null(object)) {
```

Comment:
With this change, it does not appear that `Object.get_by_full_path` can return a null object. However, it appears that other functions such as `Object.get_by_path` are still able to.

If we need to preserve `Object.get_by_full_path`'s ability to return a null object, then the `stop()` at `Dataset.get_by_full_path` should be sufficient.